### ISSUE TITLE
[BUG] `no-magic-number` lint appears for `const` Map literals that are immediately returned from functions

### ISSUE THREAD

**Issue by solid-yuriiprykhodko** [dart-code-checker/dart-code-metrics#690]
* **Dart code metrics version:** 4.10.1
* **Dart sdk version:** Dart SDK version: 2.15.1 (stable) (Tue Dec 14 13:32:21 2021 +0100) on "macos_x64"

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
dart_code_metrics:
  rules:
    - no-magic-number
```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart
// ignore_for_file: unused_element

enum Key { a, b, c }

class _StructWithMap {
  final Map<Key, double> m;

  _StructWithMap({
    required this.m,
  });
}

// lint
Map<Key, double> _mapMakerLintA() => const {
      Key.a: 0.1,
      Key.b: 0.1,
      Key.c: 0.1,
    };

// lint
Map<Key, double> _mapMakerLintB() {
  return const {
    Key.a: 0.1,
    Key.b: 0.1,
    Key.c: 0.1,
  };
}

// no lint
Map<Key, double> _mapMakerOK() {
  const m = {
    Key.a: 0.1,
    Key.b: 0.1,
    Key.c: 0.1,
  };

  return m;
}

// lint
_StructWithMap _structMakerLintA() => _StructWithMap(
      m: const {
        Key.a: 0.1,
        Key.b: 0.1,
        Key.c: 0.1,
      },
    );

// lint
_StructWithMap _structMakerLintB() {
  return _StructWithMap(
    m: const {
      Key.a: 0.1,
      Key.b: 0.1,
      Key.c: 0.1,
    },
  );
}

// no lint
_StructWithMap _structMakerOK() {
  final s = _StructWithMap(
    m: const {
      Key.a: 0.1,
      Key.b: 0.1,
      Key.c: 0.1,
    },
  );

  return s;
}
```

**What did you expect to happen?**
`no-magic-number` lint not appearing.

**What actually happened?**
`no-magic-number` lint appears for `const` Map literals that are constructed and immediately returned from functions.
This also works for Map literals embedded in other classes.

**Are you willing to submit a pull request to fix this bug?**
No.

**Comment by solid-yuriiprykhodko** [comment]
A sample project with issue repro here: https://github.com/solid-yuriiprykhodko/metrics_cc_repro/tree/magic-number-in-map

**Comment by dkrutskikh** [comment]
Available in 4.11.0 release 🚀
